### ISSUE TITLE
Datepicker doesn't work with ngIf and Angular 1.3.17

### ISSUE THREAD

**Issue by realityking** [angular-ui/bootstrap#4174]
If a datepicker is dynamically shown using Angular 1.3.17 and Angular Animate the following error is triggered:

```
TypeError: Cannot read property 'parentNode' of undefined
    at angular-animate.js:1205
    at angular-animate.js:541
    at Scope.$get.Scope.$digest (angular.js:14447)
    at Scope.$get.Scope.$apply (angular.js:14660)
    at HTMLButtonElement.<anonymous> (angular.js:21697)
    at HTMLButtonElement.eventHandler (angular.js:3047)
```

This goes back to the following commit: b72efed56f68e9e5dbed934df7f47e0b5c94242c

Plunkr: http://embed.plnkr.co/pBbrC11O9p6s46PkF7ky/preview

I've included a file called ui-bootstrap-old.js which includes the last working version (dbceec76d0d99f2b8204c0f2a5422b76f363a7f7).

I could not reproduce the issue with Angular 1.4

**Comment by RobJacobs** [comment]
I'm not seeing that error on the demo page with Angular 1.3.17 as show in this [plunk](http://embed.plnkr.co/W4meKeC1bMN1WE4r1NAK/preview).  The implementation in the plunk you linked:

    <div ng-controller="MainCtrl">
        <button type="button" ng-click="toggle()" class="btn btn-default">Toggle</button>
        <div ng-if="isShown">
            <input type="text" class="form-control input-sm" datepicker-popup="dd.MM.yyyy" ng-model="data.date" />
        </div>
    </div>

Has the div wrapping the datepicker with an ng-if which is no longer necessary unless you want to hide the input element as well?  And there is no is-open attribute on the input element with the datepicker popop.

**Comment by wesleycho** [comment]
I am seeing that error in the Plunker.

That is an ngAnimate bug in 1.3 unfortunately - the datepicker itself does not use ngAnimate.

**Comment by wesleycho** [comment]
This ticket is a duplicate of #4012.

**Comment by wesleycho** [comment]
This is now fixed in Angular in 1.3.18.
